packages/i18n/I18nDecorator/I18nDecorator.js

```diff
@@ -10,21 +10,44 @@
 import {on, off} from '@enact/core/dispatcher';
 import hoc from '@enact/core/hoc';
 import {Publisher, contextTypes as stateContextTypes} from '@enact/core/internal/PubSub';
+import {Job} from '@enact/core/util';
 import PropTypes from 'prop-types';
 import React from 'react';
 
-import ilib from '../src/index.js';
 import {isRtlLocale, updateLocale} from '../locale';
+import ilib from '../src/index.js';
+import {setResBundleLocale} from '../src/resBundle';
+import wrapIlibCallback from '../src/wrapIlibCallback';
 
 import getI18nClasses from './getI18nClasses';
 
+const join = (a, b) => a && b ? a + ' ' + b : (a || b || '');
+
 const contextTypes = {
 	rtl: PropTypes.bool,
 	updateLocale: PropTypes.func
 };
 
 const I18nContext = React.createContext(null);
 
+/**
+ * Default config for `ResolutionDecorator`.
+ *
+ * @memberof i18n/I18nDecorator.I18nDecorator
+ * @hocconfig
+ */
+const defaultConfig = {
+	/**
+	 * Retrieve i18n resource files synchronously
+	 *
+	 * @type {Boolean}
+	 * @default false
+	 * @public
+	 * @memberof i18n/I18nDecorator.I18nDecorator.defaultConfig
+	 */
+	sync: false
```

Comment:
In response to your question in the description
> Should `sync` be `true` or `false` by default for `i18n/I18nDecorator`? `false` is the direction but `true` is the current reality for Moonstone. Is setting it to `true` by default for `moonstone/MoonstoneDecorator` enough?

IMHO, I think it depends on the status of `ilib`'s async support. If the version of `ilib` used in `enact` does not support 100%, then setting `sync` to `true` would be better. If not, async is probably a preferable way to do so.